### PR TITLE
Fix cppcheck issues

### PR DESCRIPTION
**Pull request by DavidMansolino**
**Description**
  - [x] Fix new cppcheck issues on macOS.
  - [x] Add a new `test sources` label to trigger sources test on all platforms
  - [x] Fix python3.7 brew build on macOS.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
I am taking advantage of this PR to introduce a new 'test sources' label to run the test on all the platforms without switching the PR to ready for review.

**Review by omichel — APPROVED** [review]
Thank you.

File: src/webots/sound/WbPicoTextToSpeech.cpp [hunk 1]
```diff
@@ -351,6 +351,10 @@ qint16 *WbPicoTextToSpeech::generateBufferFromText(const QString &text, int *siz
         buffer = (qint16 *)realloc(buffer, bufferSize);
         if (!buffer)  // re-allocation failed, this is required otherwise CppCheck raises an error
           free(previousBuffer);
+        else {
+          gError = QString("Cannot re-allocate buffer.");
+          return NULL;
+        }
```
Comment (stefaniapedrazzi) [inline src/webots/sound/WbPicoTextToSpeech.cpp:357]: This code seems wrong (and it is still failing at cppcheck on macOS https://github.com/cyberbotics/webots/pull/2327/checks?check_run_id=1259146248 ).

If `buffer` is not NULL, then the allocation didn't fail so we should neither return nor print an error.

Comment (stefaniapedrazzi) [reply src/webots/sound/WbPicoTextToSpeech.cpp:357]: I fixed this in  #2347

Comment (DavidMansolino) [reply src/webots/sound/WbPicoTextToSpeech.cpp:357]: You're completely right, fixing this right now.
Strangely, the cppcheck did't fails my previous PRs today: https://github.com/cyberbotics/webots/runs/1257659737#step:4:45

**Pull request merged**